apis/v1alpha1/shared_types.go

```diff
@@ -175,6 +175,20 @@ type AdminNetworkPolicyEgressPeer struct {
 	// <network-policy-api:experimental>
 	// +optional
 	Nodes *metav1.LabelSelector `json:"nodes,omitempty"`
+	// Networks defines a way to select IPs and CIDR blocks that represent
+    // entities that live outside the cluster as a peer.
+	// It is the list of NetworkCIDR (both v4 & v6) that can be used to define
+	// external destinations.
+    // You could use this to select internal cluster networks like podCIDR blocks, but it is
+    // recommended to use namespaces and pods peers instead in those cases.
+	//
+	// Support: Extended
+	//
+	// <network-policy-api:experimental>
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=100
+	Networks []string `json:"networks,omitempty" validate:"omitempty,dive,cidr"`
```

Comment:
need to come back to how we want the validation of the cidr's to be done.

Reply:
using https://blog.markhatton.co.uk/2011/03/15/regular-expressions-for-ip-addresses-cidr-ranges-and-hostnames/ for now (took from ocp/api)